docs/tools.md

```diff
@@ -0,0 +1,114 @@
+# Tools
+
+Tools are Python packages that provide command-line interfaces. Tools can be invoked without installation using `uvx`, in which case their dependencies are installed in a temporary virtual environment isolated from the current project. Alternatively, tools can be installed with `uv tool install`, in which case their executables are placed in the `PATH` — an isolated virtual environment is still used but it is not treated as disposable.
```

Comment:
Some questions that come up for me after reading this:

* Does the temporary venv get deleted after `uvx` is done running?
* How does `uv tool install` choose where to put a program on my `PATH`? I think if you just put, "By using XDG" or something in parentheses, that would probably answer my question.